### PR TITLE
[prometheusreceiver] Fix metrics being grouped into the same metrics family incorrectly

### LINKED ISSUE #13117: prometheusreceiver: possible incorrect metric name for cumulative metrics
**Describe the bug**
given metrics like the following, exported metrics will only be `envoy_cluster_upstream_rq`
```
# TYPE envoy_cluster_upstream_rq counter
envoy_cluster_upstream_rq{envoy_response_code="200",envoy_cluster_name="local_envoy_admin"} 8130
envoy_cluster_upstream_rq{envoy_response_code="200",envoy_cluster_name="local_service83"} 8265
envoy_cluster_upstream_rq{envoy_response_code="200",envoy_cluster_name="sds_server_uds"} 1
envoy_cluster_upstream_rq{envoy_response_code="503",envoy_cluster_name="sds_server_uds"} 2

# TYPE envoy_cluster_upstream_rq_total counter
envoy_cluster_upstream_rq_total{envoy_cluster_name="local_envoy_admin"} 8131
envoy_cluster_upstream_rq_total{envoy_cluster_name="local_service80"} 0
envoy_cluster_upstream_rq_total{envoy_cluster_name="local_service81"} 0
envoy_cluster_upstream_rq_total{envoy_cluster_name="local_service82"} 0
envoy_cluster_upstream_rq_total{envoy_cluster_name="local_service83"} 8265
envoy_cluster_upstream_rq_total{envoy_cluster_name="sds_server_uds"} 1
```

when `envoy_cluster_upstream_rq` appears first, the family name will be `envoy_cluster_upstream_rq`, [normalizeMetricName](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b33c86d56a36875a6848bb95135d5756c9a8b3e3/receiver/prometheusreceiver/internal/util.go#L74) returns the same metric name, later when `envoy_cluster_upstream_rq_total` comes, after normalized, it will be `envoy_cluster_upstream_rq`, thus they will be [grouped](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b33c86d56a36875a6848bb95135d5756c9a8b3e3/receiver/prometheusreceiver/internal/otlp_metricsbuilder.go#L182-L188) into the same metric family 

**Steps to reproduce**
test prometheus receiver with above metrics

**What did you expect to see?**
4 `envoy_cluster_upstream_rq` metrics and 6 `envoy_cluster_upstream_rq_total` metrics

**What did you see instead?**
10 `envoy_cluster_upstream_rq` metrics.

**What version did you use?**
Version: (e.g., `v0.4.0`, `1eb551b`, etc)

**What config did you use?**
Config: (e.g. the yaml config file)

**Environment**
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

**Additional context**

### PR DESCRIPTION
**Pull request by newly12**
**Description:** <Describe what has changed.>
Fix metrics being grouped into the same metrics family incorrectly

**Link to tracking Issue:** <Issue number if applicable>
Fixes #13117 

**Testing:** <Describe what testing was performed and which tests were added.>
UT

**Documentation:** <Describe the documentation added.>
n/a

### CODE REVIEW

File: receiver/prometheusreceiver/metrics_receiver_test.go [hunk 1]
```diff
@@ -1076,6 +1076,15 @@ rpc_duration_seconds_sum{foo="no_quantile"} 101
 rpc_duration_seconds_count{foo="no_quantile"} 55
 `
 
+var target4Page1 = `
```
Comment (dashpole) [inline receiver/prometheusreceiver/metrics_receiver_test.go:1079]: Note that in OpenMetrics, the following is valid:

```
# A counter named "foo".  In OpenMetrics, the counter metric point includes _total, but the TYPE comment does not 
# TYPE foo counter
foo_total 0
# A gauge named foo_total, which also has the same name on the metric point.
# TYPE foo_total gauge
foo_total 1
```
It is [discouraged](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes) to do so, though.

We have to choose to either check the normalized or the non-normalized name first, though.  If we check normalized first, we hit your issue.  If we check non-normalized first, we hit the issue above.

**Review by dashpole — APPROVED** [review]
Overall, I think this is a less bad thing to do.

File: receiver/prometheusreceiver/metrics_receiver_test.go (hunk 1 above)
Comment (newly12) [reply receiver/prometheusreceiver/metrics_receiver_test.go:1079]: looks like so, unless we are able to know the metric type, it looks like we can only choose one issue to fix..

**Comment by newly12** [comment]
@Aneurysm9 could you please comment?

**Comment by bogdandrutu** [comment]
@dashpole please do a final review and add the ready to merge label after.

**Review by codeboten — COMMENTED** [review]
> I think we should wait on this. I was able to discuss this at the prom wg yesterday (https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13117#issuecomment-1261065219), and we concluded that we should see if it is possible for the appender to know if the metrics came from OpenMetrics or from Prometheus

@dashpole do you want to add a request for change until this is unblocked?

**Review by dashpole — CHANGES_REQUESTED** [review]
lets see if we can get a feature in upstream promethetheus that lets us know if the scrape was openmetrics or prometheus

**Comment by newly12** [comment]
Thanks @dashpole , do we have any prometheus issue being tracked for this currently?

**Comment by dashpole** [comment]
While writing up the issue, i've realized that knowing openmetrics vs prometheus won't help us.  I was going to write:

> The OpenTelemetry collector's prometheus receiver is implemented as an Appender.  One issue we currently have is that we don't know how to correctly associate metric points with type (or other) comments.  For example, with the Prometheus metrics:

```
# TYPE foo gauge
foo{} 1
# TYPE foo_total counter
foo_total{} 1
```

> Each metric point should be associated with the metric family of the same name (`foo` with `foo`, `foo_total` with `foo_total`)

> However, with the OpenMetrics metrics:
```
# TYPE foo counter
foo_total{} 1
# TYPE foo_total gauge
foo_total{} 1
```

> We want to associate the metrics with the counter, rather than the gauge.

But then I read in the [OpenMetrics spec](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes):

> The name of a MetricFamily MUST NOT result in a potential clash for sample metric names as per the ABNF with another MetricFamily in the Text Format within a MetricSet. An example would be a gauge called "foo_created" as a counter called "foo" could create a "foo_created" in the text format.

So the OpenMetrics example above is invalid.  In that case, I think it is correct for us to accept this PR.

File: receiver/prometheusreceiver/metrics_receiver_test.go (hunk 1 above)
Comment (dashpole) [reply receiver/prometheusreceiver/metrics_receiver_test.go:1079]: My assertion above is incorrect.

**Review by dashpole — APPROVED** [review]

**Comment by newly12** [comment]
@codeboten @Aneurysm9 could you please review this PR?

**Review by codeboten — APPROVED** [review]

**Pull request merged**